tinygrad/codegen/uops.py

```diff
@@ -294,14 +294,10 @@ def rewrite(u:UOp) -> UOp:
         changed += recurse_cnt
         # NOTE: this changes UOp, so we have to delete caches
         up.vin = tuple(rewrite(x) for x in up.vin)
-        try: del up.parents
-        except AttributeError: pass
-        try: del up.cmp_tuple
-        except AttributeError: pass
+        if 'parents' in up.__dict__: delattr(up, 'parents')
+        if 'cmp_tuple' in up.__dict__: delattr(up, 'cmp_tuple')
```

Comment:
hmm so we had an if here but it got removed because of being "too slow"
- is this faster if you isolate it?
- is there a perf diff between this and hasattr?

Reply:
I've updated my [benchmark with time result at the end ](https://paste.fo/50de9641ee2a).
`del up.parents ` is as fast as `delattr(up, 'parents')`  
but `hasattr` is indeed very slow as well as `getattr` (they are both used a lot in tinygrad, but we can't use `.__dict__` on every object)